src/otx/v2/adapters/torch/lightning/modules/models/visual_prompters/segment_anything.py

```diff
@@ -0,0 +1,831 @@
+
+        Returns:
+            Tensor: The DICE loss.
+        """
+        numerator = 2 * (inputs * targets).sum(-1)
+        denominator = inputs.sum(-1) + targets.sum(-1)
+        loss = 1 - (numerator + 1) / (denominator + 1)
+        return loss.sum() / num_masks
+
+    def calculate_sigmoid_ce_focal_loss(
+        self,
+        inputs: Tensor,
+        targets: Tensor,
+        num_masks: int,
+        alpha: float = 0.25,
+        gamma: float = 2,
+    ) -> Tensor:
+        r"""Loss used in RetinaNet for dense detection: https://arxiv.org/abs/1708.02002.
+
+        Args:
+            inputs (Tensor): A float tensor of arbitrary shape.
+            targets (Tensor): A tensor with the same shape as inputs. Stores the binary classification labels
+                for each element in inputs (0 for the negative class and 1 for the positive class).
+            num_masks (int): The number of masks present in the current batch, used for normalization.
+            alpha (float, *optional*, defaults to 0.25): Weighting factor in range (0,1)
+                to balance positive vs negative examples.
+            gamma (float, *optional*, defaults to 2.0): Exponent of the modulating factor \\(1 - p_t\\)
+                to balance easy vs hard examples.
+
+        Returns:
+            Tensor: The focal loss.
+        """
+        loss = functional.binary_cross_entropy_with_logits(inputs, targets, reduction="none")
+        if self.config.model.loss_type.lower() == "sam":
+            # focal loss for SAM loss
+            p_t = inputs * targets + (1 - inputs) * (1 - targets)
+            loss = loss * ((1 - p_t) ** gamma)
+            if alpha >= 0:
+                alpha_t = alpha * targets + (1 - alpha) * (1 - targets)
+                loss = alpha_t * loss
+        return loss.mean(1).sum() / num_masks
+
+    def calculate_iou(self, inputs: Tensor, targets: Tensor, epsilon: float = 1e-7) -> Tensor:
+        """Calculate the intersection over union (IOU) between the predicted mask and the ground truth mask.
+
+        Args:
+            inputs (Tensor): A tensor representing a mask.
+            targets (Tensor): A tensor with the same shape as inputs. Stores the binary classification labels
+                for each element in inputs (0 for the negative class and 1 for the positive class).
+            epsilon (float, *optional*, defaults to 1e-7): A small value to prevent division by zero.
+
+        Returns:
+            Tensor: The IOU between the predicted mask and the ground truth mask.
+        """
+        pred_mask = (inputs >= 0.5).float()
+        intersection = torch.sum(torch.mul(pred_mask, targets), dim=1)
+        union = torch.sum(pred_mask, dim=1) + torch.sum(targets, dim=1) - intersection
+        return intersection / (union + epsilon)
+
+    def export(
+        self,
+        export_dir: str | Path,
+        export_type: str = "OPENVINO",
```

Comment:
maybe we could have `export_type` as Enum-str in otx in the future.

Reply:
yes, i agree.